velox/experimental/cudf/tests/TableScanTest.cpp

```diff
@@ -439,3 +481,54 @@ TEST_F(TableScanTest, splitOffset) {
           .assertEmptyResults(),
       "CudfHiveDataSource cannot process splits with non-zero offset");
 }
+
+TEST_F(TableScanTest, s3DataSource) {
+  auto minioServer = std::make_unique<MinioServer>();
+  minioServer->start();
+
+  const auto sourceFile = facebook::velox::test::getDataFilePath(
+      "velox/connectors/hive/storage_adapters/s3fs/tests",
+      "../../../../velox/dwio/parquet/tests/examples/int.parquet");
+  const char* bucketName = "data";
+  const auto destinationFile =
+      minioServer->path() + "/" + bucketName + "/int.parquet";
+
+  minioServer->addBucket(bucketName);
+  std::ifstream src(sourceFile, std::ios::binary);
+  std::ofstream dest(destinationFile, std::ios::binary);
+  // Copy source file to destination bucket.
+  dest << src.rdbuf();
+  ASSERT_GT(dest.tellp(), 0) << "Unable to copy from source " << sourceFile;
+  dest.close();
+
+  // Reset the CudfHiveConnector config to use buffered data source
+  auto configOverride = std::unordered_map<std::string, std::string>{
+      {facebook::velox::cudf_velox::connector::hive::CudfHiveConfig::
+           kUseBufferedInput,
+       "true"}};
```

Comment:
Just to answer your offline query @karthikeyann, If we set this to `false` instead, this test will try to use kvikio to do I/O and will fail as it doesn't have all the S3 config from `minioserver->hiveConfig()`